crates/zed/src/languages/ruby/highlights.scm

```diff
@@ -70,10 +70,23 @@
   "**" @operator
 ) @constant.builtin
 
+(constant) @type
+
 ((constant) @constant
  (#match? @constant "^[A-Z\\d_]+$"))
 
-(constant) @type
+(scope_resolution
+  ((constant) @superclass
+  (#match? @type "^([A-Z][a-zA-Z]*(?:::[A-Z][a-zA-Z]*)+)$"))
+) @operator
+
+(class
+  (constant) @type
+)
+
+(superclass
+  (constant) @superclass
```

Comment:
I think we'll want to use something like `@type.super` or `@type.parent` here.

This would have the additional benefit of automatically falling back to the `type` styles in the theme if `type.super` is missing.